docs/developing-flows/message-design.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: docs-developing-flows
+toc: toc-developing-flows.html
+title: Message design
+slug: message design
+---
+
+The messages that pass through a flow are plain JavaScript objects that can have
+properties set on them.
+
+They usually have a payload property - this is the default property that most nodes
+will work with.
+
+For more information about messages in Node-RED you should read the [Working with messages](/docs/user-guide/messages)
+section of the user guide.
+
+This section looks at some of the choices you need to make when deciding how to
+stucture the messages in your flows.
+
+### Working with the payload
+
+When creating flows, the choice of properties used on a message will largely
+be determined by what the nodes in the flow require.
+
+Most nodes will expect to work with `msg.payload` and that will guide most of the choices you make.
+
+For example, consider a flow that receives an id in the payload of an MQTT message. It then uses that id to query a database to find a matching record.
+
+<div style="width: 300px" class="figure">
+  <img src="images/placeholder.png" alt="Link nodes">
+  <p class="caption">Placeholder for image: example mqtt to database query flow</p>
+</div>
+
+The database node will put its result in the payload of the message it sends - overwritting the original id value. If the flow needs to be able to reference that id value later on, it can use a Change node to copy the value to another property that will not get overwritten.
+
+<div style="width: 300px" class="figure">
+  <img src="images/placeholder.png" alt="Link nodes">
+  <p class="caption">Placeholder for image: example change node to save msg.id</p>
+</div>
+
+
+This highlights an important principle: nodes should not modify or remove properties on messages that are not related to their functionality.
+
+For example, in most cases, a Function node should send on the same message object it received rather than create a new message object.
```

Comment:
just a Function node ? (Ok I get for most users yes... but...)

Reply:
The Function node is the only node a user can explicitly create a new message object which is why it is singled out here.